docs(em): sending logs to cockpit (#2721)
scaleway/docs-content: bare-metal/elastic-metal/how-to/send-metrics-logs-to-cockpit.mdx

---
meta:
  title: How to send metrics & logs from bare metal servers to Cockpit
  description: This page explains how to send metrics & logs from bare metal servers to Cockpit
content:
  h1: How to send metrics & logs from bare metal servers to Cockpit
  paragraph: This page explains how to send metrics & logs from bare metal servers to Cockpit
tags: metrics grafana cockpit elastic-metal
dates:
  validation: 2024-02-13
  posted: 2024-02-13
categories:
  - bare-metal
---

Monitoring your dedicated servers with [Scaleway Cockpit](/observability/cockpit/quickstart/) offers detailed information about performance and health, ensuring optimal functionality and proactive issue resolution.
Cockpit provides a user-friendly interface for real-time monitoring of CPU, memory, disk usage, and network activity, enhancing the reliability and performance of bare metal servers.

> 

Reviewer: I think Maxime mentioned adding a message box to warn about additional costs that might apply. Maybe copy/paste the one this page https://www.scaleway.com/en/docs/observability/cockpit/how-to/send-metrics-logs-to-cockpit/

<Macro id="iam-requirements" />

<Message type="requirement">
  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
  - You have an Elastic Metal or Dedibox server running on a Linux Operating system
</Message>

## Activating and Accessing Cockpit

> 1. Activate Cockpit on your Scaleway account if you have not already. Follow the instructions in the [official documentation](/observability/cockpit/how-to/activate-cockpit/) for further information how to do so.

Reviewer: ```suggestion
1. [Activate Cockpit on your Scaleway account](/observability/cockpit/how-to/activate-cockpit/) if you have not already. 
```

> 2. Generate a push token to push metrics/logs to Cockpit by following the instructions provided [here](/observability/cockpit/how-to/create-token/).

Reviewer: ```suggestion
2. Generate a push token to push metrics/logs to Cockpit by following the instructions provided [in the dedicated documentation](/observability/cockpit/how-to/create-token/).
```

3. Configure Grafana Agent to push metrics/logs to Cockpit. Refer to the [official documentation](/observability/cockpit/api-cli/configuring-grafana-agent/) for detailed instructions. 
    <Message type="tip">
        Ensure to set the interval to 1 minute for real-time monitoring.
    </Message>

> 4. Retrieve Grafana credentials to access your Cockpit dashboards. Instructions can be found [here](/observability/cockpit/how-to/retrieve-grafana-credentials/).

Reviewer 2: ```suggestion
4. [Retrieve your Grafana credentials](/observability/cockpit/how-to/retrieve-grafana-credentials/) to access your Cockpit dashboards.
```

> Once you have the Grafana credentials, access your Cockpit dashboards by navigating to the Cockpit dashboard URL provided in the [official documentation](/observability/cockpit/how-to/access-grafana-and-managed-dashboards/).

Reviewer 2: Maybe we just add the link to the documentation on accessing Cockpit dashboards to step 4, as it repeats a bit in here, and delete this sentence? Something like:

`4. [Retrieve your Grafana credentials](/observability/cockpit/how-to/retrieve-grafana-credentials/), then use them to [access your Cockpit dashboards](/observability/cockpit/how-to/access-grafana-and-managed-dashboards/).`

> ## Importing Dashboards

Reviewer 2: ```suggestion
## Importing dashboards
```

To monitor your bare metal servers, you can import pre-configured dashboards from the Grafana dashboard repository. One such dashboard is the Node Exporter Full dashboard, offering detailed server metrics. Follow these steps to import it:

1. Visit the [Node Exporter Full dashboard](https://grafana.com/grafana/dashboards/12486-node-exporter-full/) on the Grafana dashboard repository.
2. Copy the dashboard ID by clicking **Copy ID**.
3. In Cockpit, navigate to the **Dashboards** section.
4. Click **Import** and paste the copied dashboard ID.
5. Follow the prompts to complete the import process.

> Following these steps enables you to push metrics/logs to Scaleway Cockpit (Grafana) and set up monitoring for your Scaleway Elastic Metal servers, allowing real-time tracking of server performance and health.

Reviewer 2: ```suggestion
This process allows you to push metrics and logs to Scaleway’s Cockpit (Grafana) and monitor your Elastic Metal servers with real-time tracking of their health and performance.
```

</Message>

<Message type="important">
* Having the default configuration on your agents might lead to more of your resources’ metrics being sent, a high consumption and a high bill at the end of the month.
* Sending metrics and logs for Scaleway resources or personal data using an external path is a billable feature.  In addition, any data that you push yourself is billed, even if you send data from Scaleway products. Refer to the [product pricing](https://www.scaleway.com/en/pricing/?tags=available,managedservices-observability-cockpit) for more information.
</Message>

## Activating and accessing Cockpit

1. [Activate Cockpit on your Scaleway account](/observability/cockpit/how-to/activate-cockpit/) if you have not already. 
2. Generate a push token to push metrics/logs to Cockpit by following the instructions provided [in the dedicated documentation](/observability/cockpit/how-to/create-token/).
3. Configure Grafana Agent to push metrics/logs to Cockpit. Refer to the [official documentation](/observability/cockpit/api-cli/configuring-grafana-agent/) for detailed instructions. 
    <Message type="tip">

>         Ensure to set the interval to 1 minute for real-time monitoring.

Author: ```suggestion
        For efficient cost management and near real-time resource monitoring, it is recommended to set the interval to 1 minute.
```
